src/SFML/Audio/SoundFileReaderFlac.cpp

```diff
@@ -142,10 +138,12 @@ namespace
                         break;
                 }
 
+                // If there's room in the output buffer, copy the sample there
                 if (data->remaining > 0)
                 {
-                    // There's room in the output buffer, copy the sample there
-                    *data->buffer++ = sample;
+                    // If there's no output buffer, it means that we are seeking, so just skip the output write
+                    if (data->buffer)
+                        *data->buffer++ = sample;
                     data->remaining--;
```

Comment:
Previously, the whole function returned early when `data->buffer` was null. Since I removed that check, I figured I'd put a smaller one around the point where the buffer is actually dereferenced. I know that in practice, `data->remaining` will always be 0 when `data->buffer` is null (since that's the seek condition), and the `data->leftovers.push_back(sample)` path should always be taken then. But I'm always wary of dereferencing a pointer that hasn't been checked locally. So I just covered the single dereferencing line of code with the new check. I didn't touch the decrement because that preserves the countdown to where it will start loading the samples into the always-valid leftover buffer. But that's a bit of moot point since that should never happen anyway.

Maybe I should change the surrounding if condition to `if (data->buffer && data->remaining > 0)`.